src/components/ui/Toaster/Toast/Toast.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { HTMLAttributes, useEffect, useState } from 'react';
+import { block } from 'bem-cn';
+
+import { Button, Icon } from '@/components/ui';
+
+import { ToastItem } from './types';
+
+import closeSvg from 'bootstrap-icons/icons/x-lg.svg';
+
+type ToastProps = {
+  toast: ToastItem;
+  deleteToast: (id: string) => void;
+} & HTMLAttributes<HTMLDivElement>;
+
+const b = block('toast');
+
+function Toast(props: ToastProps): JSX.Element {
+  const { toast, deleteToast } = props;
+  const [timer, setTimer] = useState<NodeJS.Timeout>();
+  const [startTimer, setStartTimer] = useState(0);
+  const [leftTimerTime, setLeftTimerTime] = useState(0);
+
+  useEffect(() => {
+    setStartTimer(performance.now());
+    const timeout = setTimeout(() => {
+      if (toast.timeout) {
+        deleteToast(toast.id);
+      }
+    }, toast.timeout || 1);
+
+    setTimer(timeout);
+
+    return () => {
+      clearTimeout(timer as NodeJS.Timeout);
+    };
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [toast.timeout]);
+
+  function onMouseEnter() {
+    if (toast.timeout) {
+      setLeftTimerTime(5000 - (performance.now() - startTimer));
```

Comment:
❗️
Хардкода быть не должно)
```suggestion
      setLeftTimerTime(timer.timeout - (performance.now() - startTimer));
```